climate-monitoring-client/src/main/java/climatemonitoring/Login.java

```diff
@@ -1,13 +1,39 @@
 package climatemonitoring;
 
+import climatemonitoring.core.ConnectionLostException;
+import climatemonitoring.core.DatabaseRequestException;
+import climatemonitoring.core.Operator;
 import climatemonitoring.core.ViewState;
+import climatemonitoring.core.headless.Console;
 
 class Login extends ViewState {
 
 	@Override
 	public void onHeadlessRender(String args) {
 
-		throw new UnsupportedOperationException("Unimplemented method 'onHeadlessRender'");
+		String[] parts = args.split(" ");
+		
+		if(parts[0] == null || parts[1] == null)
+			Console.write("inserire username o password");
```

Comment:
You should check if the array is the correct size (2 in this case), and if it is not, write an error message and then terminate the procedure + you need to write in english